modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkIndexByScrollAction.java

```diff
@@ -86,6 +86,9 @@ protected BulkRequest buildBulk(Iterable<SearchHit> docs) {
         Map<String, Object> scriptCtx = null;
 
         for (SearchHit doc : docs) {
+            if (doc.isSourceEmpty()) {
+                throw new IllegalArgumentException("[" + doc.index() + "][" + doc.type() + "][" + doc.id() + "] didn't store _source");
```

Comment:
Is the document having an empty source the same as a _source being disabled. Presumably I could have _source enabled on my index but I happen to have some documents which have no content (e.g. dirty data or a failed etl process on that document). Should we not actually check the mapping to determine if _source is disabled rather than letting one bad document fail an entire reindex?